velox/connectors/hive/storage_adapters/s3fs/S3FileSystem.cpp

```diff
@@ -115,14 +151,10 @@ class S3ReadFile final : public ReadFile {
     std::stringstream ss;
     ss << "bytes=" << offset << "-" << offset + length - 1;
     request.SetRange(awsString(ss.str()));
-    // TODO: Avoid copy below by using  req.SetResponseStreamFactory();
-    // Reference: ARROW-8692
+    request.SetResponseStreamFactory(
+        AwsWriteableStreamFactory(reinterpret_cast<char*>(position), length));
```

Comment:
position is already char*; what's the reason for applying reinterpret_cast ?

Reply:
No need. Fixed.